Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/InstallDiffCommand.cs

```diff
@@ -516,21 +533,34 @@ private void AddDirectory(string dirName, CancellationToken cancellationToken)
             _modifiedFilesStatusReporter.IsActive.Value = true;
             _modifiedFilesStatusReporter.Description.Value = "Applying diffs...";
 
-            for (int i = 0; i < _diffSummary.ModifiedFiles.Length; i++)
+            var nameFiles = _pack1Meta.Files.Select(f => f.Name).Where(f => !_diffSummary.AddedFiles.Contains(f))
+                .ToArray();
+            for (int i = 0; i < nameFiles.Length; i++)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                var entryName = _diffSummary.ModifiedFiles[i];
+                var entryName = nameFiles[i];
 
                 if (!entryName.EndsWith("/"))
                 {
                     PatchFile(entryName, packageDirPath, suffix, tempDiffDir, cancellationToken, i);
                 }
 
-                _modifiedFilesStatusReporter.Progress.Value = (i + 1) / (double) _diffSummary.ModifiedFiles.Length;
-                _modifiedFilesStatusReporter.Description.Value = string.Format("Applying diffs ({0}/{1})...", i + 1, _diffSummary.ModifiedFiles.Length);
+                _modifiedFilesStatusReporter.Progress.Value = (i + 1) / (double) nameFiles.Length;
+                _modifiedFilesStatusReporter.Description.Value =
+                    string.Format("Applying diffs ({0}/{1})...", i + 1, nameFiles.Length);
+            }
+            _localMetaData.SaveData();
+            
+            _logger.LogDebug("Register entry for unchanged files.");
+            _modifiedFilesStatusReporter.Description.Value = "Applying diffs...";
+
+            foreach (string fileName in _diffSummary.UnchangedFiles)
+            {
+                _localMetaData.RegisterEntry(fileName, _versionId, EntriesSize);
```

Comment:
Seems like a hack to me (faking the entry size), I'd propose adding a possibility for setting the size as null, and using the file counter inside `LocalMetaData` to determine when to save.

Reply:
done